### PR TITLE
Add profile section

### PR DESCRIPTION
**Pull request by supthefuhrer1**

### CODE REVIEW

File: src/app/profile/page.tsx [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+  Select,
+  Button,
+  Divider,
+} from '@mantine/core';
+import Image from 'next/image';
+import images from '@/themes/images';
+import camera from '../../assets/images/Camera.svg';
+import copy from '../../assets/images/CopySimple.svg';
+import edit from '../../assets/images/PencilSimpleLine.svg';
+import {Grid} from '@mantine/core';
```
Comment (shelly19gkmit) [inline src/app/profile/page.tsx:18]: use absolute path for assets

File: src/app/profile/page.tsx [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+  width?: number | string;
+}
+
+interface ConnectBodyProps {
+  isSocialConnected: boolean;
+  socialId: string;
+  divider?: boolean;
+  label: string;
+  setIsSocialConnected: (isConnected: boolean) => void,
+}
```
Comment (shelly19gkmit) [inline src/app/profile/page.tsx:35]: Create a separate file for maintaining types in Profile folder.

File: src/app/profile/page.tsx [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+            <Grid.Col span={12}>
+              <TextInput variant="default" placeholder="Last Name" />
+            </Grid.Col>
+            <Grid.Col span={12}>
+              <Select
+                variant="default"
+                placeholder="Gender"
+                data={['Male', 'Female']}
+              />
+            </Grid.Col>
```
Comment (shelly19gkmit) [inline src/app/profile/page.tsx:123]: Create reusable Select component

File: src/app/profile/page.tsx [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+          justify={'space-between'}
+          align={'center'}>
+          <Text c="white" size="md">
+            {label}
+          </Text>
+          {isSocialConnected ? (
+            <Text classNames={connectedTagStyles}>Connected</Text>
+          ) : (
+            <Button onClick={() => setIsSocialConnected(true)} variant="default">Connect</Button>
+          )}
```
Comment (shelly19gkmit) [inline src/app/profile/page.tsx:245]: create function

File: src/app/profile/page.tsx [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+              label={SOCIAL.GOOGLE}
+              isSocialConnected={isGoogleConnected}
+              setIsSocialConnected={setIsGoogleConnected}
+              socialId={googleId}
+            />
+            <ConnectBody
+              label={SOCIAL.METAMASK}
+              divider={false}
+              isSocialConnected={isMetaMaskConnected}
+              setIsSocialConnected={setIsMetaMaskConnected}
```
Comment (shelly19gkmit) [inline src/app/profile/page.tsx:184]: Divide Profile components into 4 different components->
Profile -> profile
Username -> profile/username
PROFILE DETAILS -> profile/details
CONNECT ACCOUNTS -> profile/connectAccounts

Create array and maintain these logics like this
const socialConnections = [
  {
    label: SOCIAL.TWITTER,
    isSocialConnected: isTwitterConnected,
    setIsSocialConnected: setIsTwitterConnected,
    socialId: twitterId,
  },
  {
    label: SOCIAL.DISCORD,
    isSocialConnected: isDiscordConnected,
    setIsSocialConnected: setIsDiscordConnected,
    socialId: discordId,
  },
  {
    label: SOCIAL.GOOGLE,
    isSocialConnected: isGoogleConnected,
    setIsSocialConnected: setIsGoogleConnected,
    socialId: googleId,
  },
  {
    label: SOCIAL.METAMASK,
    isSocialConnected: isMetaMaskConnected,
    setIsSocialConnected: setIsMetaMaskConnected,
    socialId: metamaskId,
    divider: false,
  },
];

const renderedConnections = socialConnections.map(({ label, isSocialConnected, setIsSocialConnected, socialId, divider = true }, index) => (
  <SocialConnect
    key={index}
    label={label}
    isSocialConnected={isSocialConnected}
    setIsSocialConnected={setIsSocialConnected}
    socialId={socialId}
    divider={divider}
  />
));

File: src/components/CountryPicker/index.tsx [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    <Select
+      withCheckIcon={false}
+      rightSection={<></>}
+      classNames={{
+        wrapper: styles['picker-wrapper'],
+        input: styles['picker-input'],
+        dropdown: styles['picker-dropdown'],
+        option: styles['picker-option'],
+      }}
+      //   variant='default'
```
Comment (shelly19gkmit) [inline src/components/CountryPicker/index.tsx:27]: remove constant

File: src/components/CountryPicker/index.tsx [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+  return (
+    <Select
+      withCheckIcon={false}
+      rightSection={<></>}
+      classNames={{
+        wrapper: styles['picker-wrapper'],
+        input: styles['picker-input'],
+        dropdown: styles['picker-dropdown'],
+        option: styles['picker-option'],
+      }}
```
Comment (shelly19gkmit) [inline src/components/CountryPicker/index.tsx:26]: define styles separately and use it here

File: src/components/CountryPicker/index.tsx [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+      //   variant='default'
+      allowDeselect={false}
+      value={selectedCountry}
+      onChange={(value, _option) => onChangeHandler(value)}
+      data={countryCodes.map((item, _index) => ({
+        label: `${item.flag} ${item.dial_code}`,
+        value: item.code,
+      }))}
+      disabled={readOnly}
```
Comment (shelly19gkmit) [inline src/components/CountryPicker/index.tsx:35]: maintain function

File: src/utils/helpers.tsx [hunk 9]
```diff
@@ -1,3 +1,15 @@
...
+import styles from '@/styles/theme.module.css';
+
+export const renderOption: SelectProps['renderOption'] = ({option}) => {
+  return (
+    <Flex className={styles.selectOptionRender} flex={1}>
+      <Text c={COLORS.WHITE}>{option.label}</Text>
+    </Flex>
+  );
+};
+
```
Comment (shelly19gkmit) [inline src/utils/helpers.tsx:12]: Helper file shouldn't be tsx, it should only contain logics not any components. Plz make this a separate component in components folder

**Review by shelly19gkmit — CHANGES_REQUESTED** [review]
Refactoring is needed, plz resolve the comments

File: src/components/CountryPicker/index.tsx (hunk 7 above)
Comment (supratimmalakar) [reply src/components/CountryPicker/index.tsx:26]: Already refactored and fixed the Country Picker component in the Login modal pr. Will take a pull once that pr is merged

File: src/themes/colors.ts [hunk 10]
```diff
@@ -13,6 +14,7 @@ const COLORS = {
   KGEN_ACID_GREEN: '#8CFF05',
   KGEN_SLATE_BLUE: '#1F3338',
   WHITE: '#E8F3F5',
+  DIVIDER: '#415F6666',
   ['WHITE OP 0.4']: 'rgba(232, 243, 245, 0.4)', // OP : Opacity
```
Comment (mandeepatindigg) [inline src/themes/colors.ts:18]: duplicated devider

File: src/context/walletConnect.tsx [hunk 11]
```diff
@@ -0,0 +1,47 @@
+'use client';
+
+import React, {ReactNode} from 'react';
+import {projectId, config} from '@/config/walletConnect';
+import {createWeb3Modal} from '@web3modal/wagmi/react';
+
+import {QueryClient, QueryClientProvider} from '@tanstack/react-query';
```
Comment (shivams10) [inline src/context/walletConnect.tsx:7]: We will be using RTK query and not tanstack query pls remove it

**Review by shivams10 — CHANGES_REQUESTED** [review]

**Pull request merged**